akka-projection-core/src/test/scala/akka/projection/internal/metrics/LifecycleMetricSpec.scala

```diff
@@ -39,8 +39,10 @@ class LifecycleMetricAtLeastOnceSpec extends LifecycleMetricSpec {
         runInternal(tt.projectionState) {
           instruments.offsetsSuccessfullyCommitted.get should be(numOfEnvelopes)
         }
-        instruments.startedInvocations.get should be(1)
-        instruments.stoppedInvocations.get should be(1)
+        eventually {
+          instruments.startedInvocations.get should be(1)
+          instruments.stoppedInvocations.get should be(1)
```

Comment:
There was a race condition between the stopping in `runInternal` and the assertion `instruments.stoppedInvocations.get should be(1)`. The [future returned by `stop()` completes before the `KillSwitch.shutdown()`](https://github.com/akka/akka-projection/blob/600b1c5abc21cc652c0597e06ce9637313666f4b/akka-projection-core/src/test/scala/akka/projection/internal/metrics/tools/InternalProjectionStateMetricsSpec.scala#L226) completed propagating (which sets instruments.stoppedInvocations).